lib/model/QubitDigitalObject.php

```diff
@@ -2757,7 +2757,36 @@ public static function convertVideoToMp4($originalPath, $newPath, $width = null,
             return false;
         }
 
-        $command = 'ffmpeg -y -i '.$originalPath.' -ar 44100 -c:v libx264 -pix_fmt yuv420p -c:a aac -movflags +faststart '.$newPath.' 2>&1';
+        // Read stream information
+        $format = self::readStreamInformation($originalPath);
+
+        error_log('Stream information: ' . json_encode($format, JSON_PRETTY_PRINT));
+
+        $videoCodec = isset($format['video']['codec_name']) ? $format['video']['codec_name'] : null;
+        $pixelFormat = isset($format['video']['pix_fmt']) ? $format['video']['pix_fmt'] : null;
+        $audioCodec = isset($format['audio']['codec_name']) ? $format['audio']['codec_name'] : null;
+        $sampleRate = isset($format['audio']['sample_rate']) ? $format['audio']['sample_rate'] : null;
```

Comment:
I would suggest using these variables to make a boolean argument for re-encoding the video or re-encoding the audio.

Another thing that would require a re-encode is fixing the file type. For example, you could have a file with the proper video and audio encoding, but the file is a `.mkv`, not a `.mp4`. I would recommend figuring out if `ffprobe` can output the file extension and use its output. Here I've used the `pathinfo` function which could be used if FFprobe doesn't output the file extension.

```suggestion
        $videoCodec = isset($format['video']['codec_name']) ? $format['video']['codec_name'] : null;
        $pixelFormat = isset($format['video']['pix_fmt']) ? $format['video']['pix_fmt'] : null;
        $audioCodec = isset($format['audio']['codec_name']) ? $format['audio']['codec_name'] : null;
        $sampleRate = isset($format['audio']['sample_rate']) ? $format['audio']['sample_rate'] : null;
        
        $reencodeVideo = ($videoCodec !== 'h264' || $pixelFormat !== 'yuv420p');
        $reencodeAudio = ($audioCoded !== 'aac' || $sampleRate !== 44100);
        $fixContainer = strtolower(pathinfo($originalPath, PATHINFO_EXTENSION)) !== 'mp4';
```

Reply:
> I would suggest using these variables to make a boolean argument for re-encoding the video or re-encoding the audio.

I agree, I've addressed that here: https://github.com/danloveg/atom/pull/1/commits/1133f0520b93e759fd60d9a8d31919b311709b47

> Another thing that would require a re-encode is fixing the file type. For example, you could have a file with the proper video and audio encoding, but the file is a `.mkv`, not a `.mp4`. I would recommend figuring out if `ffprobe` can output the file extension and use its output. Here I've used the `pathinfo` function which could be used if FFprobe doesn't output the file extension.

That's a good idea. Some files might have a non-MP4 file extension with a MP4 container, or an MP4 extension with the wrong container, but in both cases have the right video and audio encoding. I've made a method that checks whether both criterias are met (right extension and right container) here: https://github.com/danloveg/atom/pull/1/commits/1133f0520b93e759fd60d9a8d31919b311709b47

Reply:
I renamed that method to something more fitting here: 
https://github.com/danloveg/atom/pull/1/commits/6156ab4ac12641f9b5a4520c21bffb124a0f70db